stable/ambassador/templates/deployment.yaml

```diff
@@ -95,8 +95,12 @@ spec:
             - name: STATSD_HOST
               value: "localhost"
             {{- end }}
+            {{- if .Values.ambassador }}
             - name: AMBASSADOR_ID
               value: {{ .Values.ambassador.id | quote }}
+            {{- end }}
+            - name: AMBASSADOR_SINGLE_NAMESPACE
+              value: {{.Values.namespace.single | quote}}
```

Comment:
This can already be set by setting value:

```
env: 
 AMBASSADOR_SINGLE_NAMESPACE: true
```

We should `namespace.single` isn't in the read me and wasn't used so we should porbably remove it from `values.yaml`

It's already documented in [README.md](https://github.com/helm/charts/blob/master/stable/ambassador/README.md#single-namespace) how to migrate :)

Reply:
Okay that makes sense to me. It being left there made the configuration change less obvious. Removing it seems to be the best course of action.

Reply:
Along the same lines, I think it makes sense to move `AMBASSADOR_ID` configuration to the `env` section since it is either set or defaults to `default` if not set. It makes the template logic simpler and more cohesive to have all environment variable configurations be delegated to `env` unless otherwise prohibitive (like `AMBASSADOR_NAMESPACE`).

Thoughts?

Reply:
It makes sense, but it would be a breaking change so chart version -> 2.0.0 then :) in case people are already using `ambassador.id`. Didn't know it was set defaulted by ambassador so thats why it had an explicit value.

Reply:
Okay I am going to go down that route then. I think it makes more sense to set as many environment variables in `env` as we can.